recipes/mimalloc/all/conanfile.py

```diff
@@ -0,0 +1,138 @@
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        if self.settings.compiler == "Visual Studio":
+            self._cmake.generator = "NMake Makefiles"
+        self._cmake.definitions["CMAKE_BUILD_TYPE"] = self.settings.build_type
+        self._cmake.definitions["MI_BUILD_TESTS"] = False
+        self._cmake.definitions["MI_BUILD_SHARED"] = self.options.shared == True
+        self._cmake.definitions["MI_BUILD_STATIC"] = self.options.shared == False
+        self._cmake.definitions["MI_BUILD_OBJECT"] = self.options.shared == "object"
+        self._cmake.definitions["MI_SECURE"] = "ON" if self.options.secure else "OFF"  # Needs to be ON/OFF!
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+        if self.settings.compiler == "Visual Studio" and self.settings.arch == "x86":
+            tools.replace_path_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                                       "mimalloc-redirect.lib", "mimalloc-redirect32.lib")
+        with tools.vcvars(self.settings) if self.settings.compiler == "Visual Studio" else tools.no_op():
+            cmake = self._configure_cmake()
+            cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        with tools.vcvars(self.settings) if self.settings.compiler == "Visual Studio" else tools.no_op():
+            cmake = self._configure_cmake()
+            cmake.install()
+
+        if self.settings.os == "Windows":
+            if self.options.shared == True or self.options.shared == "object":
+                if self.settings.arch == "x86_64":
+                    self.copy("mimalloc-redirect.dll", src=os.path.join(self._source_subfolder, "bin"), dst=self._install_prefix)
+                elif self.settings.arch == "x86":
+                    self.copy("mimalloc-redirect32.dll", src=os.path.join(self._source_subfolder, "bin"), dst=self._install_prefix)
+
+        tools.rmdir(os.path.join(self.package_folder, self._install_prefix, "cmake"))
+
+    @property
+    def _install_prefix(self):
+        version = tools.Version(self.version)
+        return os.path.join("lib", "{}-{}.{}".format(self.name, version.major, version.minor))
+
+    @property
+    def _obj_name(self):
+        name = "mimalloc"
+        if self.options.secure:
+            name += "-secure"
+        if self.settings.build_type not in ("Release", "RelWithDebInfo", "MinSizeRel"):
+            name += "-{}".format(str(self.settings.build_type).lower())
+        return name
+
+    @property
+    def _lib_name(self):
+        name = "mimalloc"
+        if self.options.shared == False:
```

Comment:
Why do they do these os-dependent shenanigans.
```suggestion
        if self.settings.os == "Windows" and self.options.shared == False:
```